File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/common/domain/CellsGameState.java
package com.zwstudio.logicpuzzlesandroid.common.domain;

public class CellsGameState<G extends CellsGame<G, GM, GS>, GM, GS extends GameState> extends GameState {
    public G game;

    public Position size() {return game.size;}
    public int rows() {return game.rows();}
    public int cols() {return game.cols();}
    public boolean isValid(int row, int col) { return game.isValid(row, col); }
    public boolean isValid(Position p) {
        return game.isValid(p);
    }

    public CellsGameState(G game) {
        this.game = game;
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/common/domain/CellsGameState.kt
package com.zwstudio.logicpuzzlesandroid.common.domain

open class CellsGameState<G : CellsGame<G, GM, GS>, GM, GS : GameState>(var game: G) : GameState() {
    fun size() = game.size
    fun rows() = game.rows()
    fun cols() = game.cols()
    fun isValid(row: Int, col: Int) = game.isValid(row, col)
    fun isValid(p: Position) = game.isValid(p)
}